test/controllers/offer.controller.test.js

```diff
@@ -1,677 +1,815 @@
+            .end((err, res) => {
+              expect(res).to.have.status(412);
+              expect(res.body.success).to.be.eql(false);
+              expect(res.body.message).to.be.eql('Validation Error');
+              expect(res.body.error).to.be.eql('"Delivery State" is not allowed to be empty');
+              done();
+            });
+        });
+      });
+      context('when total amount payable is empty', () => {
+        it('returns an error', (done) => {
+          const offer = OfferFactory.build({ totalAmountPayable: '' });
+          request()
+            .post('/api/v1/offer/create')
+            .set('authorization', adminToken)
+            .send(offer)
+            .end((err, res) => {
+              expect(res).to.have.status(412);
+              expect(res.body.success).to.be.eql(false);
+              expect(res.body.message).to.be.eql('Validation Error');
+              expect(res.body.error).to.be.eql('"Total Amount Payable" must be a number');
+              done();
+            });
+        });
+      });
+      context('when allocation in percentage is empty', () => {
+        it('returns an error', (done) => {
+          const offer = OfferFactory.build({ allocationInPercentage: '' });
+          request()
+            .post('/api/v1/offer/create')
+            .set('authorization', adminToken)
+            .send(offer)
+            .end((err, res) => {
+              expect(res).to.have.status(412);
+              expect(res.body.success).to.be.eql(false);
+              expect(res.body.message).to.be.eql('Validation Error');
+              expect(res.body.error).to.be.eql('"Allocation In Percentage" must be a number');
+              done();
+            });
+        });
+      });
+      context('when title is empty', () => {
+        it('returns an error', (done) => {
+          const offer = OfferFactory.build({ title: '' });
+          request()
+            .post('/api/v1/offer/create')
+            .set('authorization', adminToken)
+            .send(offer)
+            .end((err, res) => {
+              expect(res).to.have.status(412);
+              expect(res.body.success).to.be.eql(false);
+              expect(res.body.message).to.be.eql('Validation Error');
+              expect(res.body.error).to.be.eql('"Title" is not allowed to be empty');
+              done();
+            });
+        });
+      });
+      context('when expires is empty', () => {
+        it('returns an error', (done) => {
+          const offer = OfferFactory.build({ expires: '' });
+          request()
+            .post('/api/v1/offer/create')
+            .set('authorization', adminToken)
+            .send(offer)
+            .end((err, res) => {
+              expect(res).to.have.status(412);
+              expect(res.body.success).to.be.eql(false);
+              expect(res.body.message).to.be.eql('Validation Error');
+              expect(res.body.error).to.be.eql('"Expires" must be a valid date');
```

Comment:
Can we change this to reflect an easier to understand sentence. Please update the schema as well.
```suggestion
              expect(res.body.error).to.be.eql('"Expiry Date" must be a valid date');
```